core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -78,7 +82,12 @@ public static void dropTableData(FileIO io, TableMetadata metadata) {
 
     // run all of the deletes
 
-    deleteFiles(io, manifestsToDelete);
+    boolean gcEnabled = PropertyUtil.propertyAsBoolean(metadata.properties(), GC_ENABLED, GC_ENABLED_DEFAULT);
+
+    if (gcEnabled) {
+      // delete data files only if GC is enabled
```

Comment:
This comment could be better to explain what the GC property is doing.

Reply:
Fixed.